Source/WebKit/UIProcess/WebBackForwardList.h

```diff
@@ -96,11 +99,17 @@ class WebBackForwardList : public API::ObjectImpl<API::Object::Type::BackForward
 
     void didRemoveItem(WebBackForwardListItem&);
 
+    void goToItemInternal(WebBackForwardListItem&, std::optional<size_t>& indexToUpdate);
+
+    std::optional<size_t>& provisionalOrCurrentIndex() { return m_provisionalIndex ? m_provisionalIndex : m_currentIndex; }
+    std::optional<size_t> provisionalOrCurrentIndex() const { return m_provisionalIndex ? m_provisionalIndex : m_currentIndex; }
```

Comment:
This looks like a mistake.  Could you make there be only one and be const and return a non-reference?  If you need a setter, make something that is more clearly a setter.